soh/soh/Enhancements/mods.cpp

```diff
@@ -486,6 +487,159 @@ void RegisterBonkDamage() {
     });
 }
 
+void RegisterChestapalooza() {
+    srand((unsigned)time(NULL));
+    static int roll;
+    static int statusTimer = -1;
+    static int eventTimer = -1;
+
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnItemReceive>([](GetItemEntry itemEntry) {
+        if (!CVarGetInteger("gChestTraps", 0)) {
+            return;
+        }
+        if (itemEntry.itemId == RG_ICE_TRAP) {
+            roll = 1 + rand() % 10;
+            switch (roll) {
+                case 1:
+                    if (!CVarGetInteger("gChestIce", 0)) {
+                        GameInteractor::RawAction::FreezePlayer();
```

Comment:
In addition to that, you could then handle the determination of what effect to apply with a unified function `GameInteractor::RawAction::PlayerEffect` or maybe `DamageEffect`, not sure. This would then just accept an effect enum value and apply the correct one internally. This is more of a suggestion, and probably down to preference, but other devs who have worked with this project more could probably chime in.